Dockerfile.rocm

```diff
@@ -102,13 +102,20 @@ ENV RAY_EXPERIMENTAL_NOSET_ROCR_VISIBLE_DEVICES=1
 
 ENV VLLM_NCCL_SO_PATH=/opt/rocm/lib/librccl.so
 
+# TODO: remove cp .so files and refactor for CI purposes
 RUN --mount=type=cache,target=/root/.cache/pip \
     pip install -U -r requirements-rocm.txt \
-    && patch /opt/rocm/include/hip/amd_detail/amd_hip_bf16.h ./rocm_patch/rocm_bf16.patch \
+    && if [ "$BASE_IMAGE" = "rocm/pytorch:rocm6.0_ubuntu20.04_py3.9_pytorch_2.1.1" ]; then \
+       patch /opt/rocm/include/hip/amd_detail/amd_hip_bf16.h ./rocm_patch/rocm_bf16.patch; fi \
     && python3 setup.py install \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_C.cpython-39-x86_64-linux-gnu.so vllm/ \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_punica_C.cpython-39-x86_64-linux-gnu.so vllm/ \
-    && cp build/lib.linux-x86_64-cpython-39/vllm/_moe_C.cpython-39-x86_64-linux-gnu.so vllm/ \
+    && if [ "$BASE_IMAGE" = "rocm/pytorch:rocm6.0_ubuntu20.04_py3.9_pytorch_2.1.1" ]; then \
+       cp build/lib.linux-x86_64-cpython-39/vllm/_C.cpython-39-x86_64-linux-gnu.so vllm/ \
+       && cp build/lib.linux-x86_64-cpython-39/vllm/_moe_C.cpython-39-x86_64-linux-gnu.so vllm/ \
+       && cp build/lib.linux-x86_64-cpython-39/vllm/_punica_C.cpython-39-x86_64-linux-gnu.so vllm/; \
+       else \
+       cp build/lib.linux-x86_64-cpython-310/vllm/_C.cpython-310-x86_64-linux-gnu.so vllm/ \
+       && cp build/lib.linux-x86_64-cpython-310/vllm/_moe_C.cpython-310-x86_64-linux-gnu.so vllm/ \
+       && cp build/lib.linux-x86_64-cpython-310/vllm/_punica_C.cpython-310-x86_64-linux-gnu.so vllm/; fi \
```

Comment:
nit: What about this?
```suggestion
        && cp build/lib.linux-x86_64-cpython-${PYTHON_VERSION}/vllm/_C.cpython-${PYTHON_VERSION}-x86_64-linux-gnu.so vllm/ \
       && cp build/lib.linux-x86_64-cpython-${PYTHON_VERSION}/vllm/_moe_C.cpython-${PYTHON_VERSION}-x86_64-linux-gnu.so vllm/ \
       && cp build/lib.linux-x86_64-cpython-${PYTHON_VERSION}/vllm/_punica_C.cpython-${PYTHON_VERSION}-x86_64-linux-gnu.so vllm/; \
```
after defining `PYTHON_VERESION`

Reply:
Good suggestion (I was thinking about it, but was thinking of doing some refactoring in the different PR). There is another way of avoiding do these copies. I thought the reason for these copies is to avoid the exception when running code (for example running unit tests in `tests` folder) in the workplace. We can use symbolic link also.